### PR TITLE
not very basic maintenance

### PR DESCRIPTION
**Pull request by faddat**
This PR originally set out to do basic maintenance on Juno.

Trouble is that tests would fail at random.

* Decided to upgrade interchaintest
* Interchaintest's changes required changes to Juno's interchain tests

....so this PR upgrades numerous libraries used in Juno, as well as its test suite.

### CODE REVIEW

**Comment by faddat** [comment]
I think that that e2e is just flaky, you probably need to rerun CI in order for this to pass.

**Comment by JakeHartnell** [comment]
Looks good, any thoughts on the failing test?

**Comment by faddat** [comment]
just should need to click the button that runs it again.

**Comment by faddat** [comment]
oops this time it was feepay

**Comment by faddat** [comment]
oh snap now it is ibchooks

let's do a 5th

**Comment by faddat** [comment]
> Looks good, any thoughts on the failing test?

If we follow the same approach as is use din the ibctest upstream, and keep spamming with commits that change nothing, it should eventually pass.

Of course it means that all of the failures were meaningless....

**or were they**?

**Comment by faddat** [comment]
https://github.com/strangelove-ventures/interchaintest/pull/1132

Screenshot 2024-05-14 at 8 54 54 AM


Look familiar?

**Comment by faddat** [comment]
Ah, pfm doesn't work again

**Comment by faddat** [comment]
Remember changes to comments do not actually change what the code does, what you're seeing is the level of flakiness in the IC tests

**Comment by faddat** [comment]
Clock again

**Comment by faddat** [comment]
"but sometimes clock doesn't poop the bed"

Yes

**Comment by faddat** [comment]
Pfm too

**Comment by faddat** [comment]
good news is that it is _probably_ not the modules themselves, just the code that tests them

**Comment by faddat** [comment]
oop, ictest-basic this time

**Comment by faddat** [comment]
go.mod is much cleaner now anyhow

**Comment by faddat** [comment]
oops tests need to be rewritten

**Comment by faddat** [comment]
```
go: downloading modernc.org/memory v1.6.0
go: downloading github.com/remyoudompheng/bigfft v0.0.0-20230129092748-24d4a6f8daec
# github.com/CosmosContracts/juno/tests/interchaintest/helpers
Error: helpers/gov.go:41:60: invalid operation: height + searchHeightDelta (mismatched types int64 and uint64)
Error: helpers/gov.go:41:86: cannot use proposalID (variable of type string) as int64 value in argument to cosmos.PollForProposalStatus
Error: helpers/gov.go:41:105: undefined: cosmos.ProposalStatusPassed
FAIL	github.com/CosmosContracts/juno/tests/interchaintest [build failed]
FAIL
make: *** [Makefile:170: ictest-feepay] Error 1
Error: Process completed with exit code 2.
```

**Comment by faddat** [comment]
* https://github.com/strangelove-ventures/interchaintest/commit/735fd2fe26f3353285e5cc1dd1527e0adcc72ad8

**Comment by faddat** [comment]
@JakeHartnell tl;dr here:

1) interchaintest is pretty unstable and will throw unanticipated results
2) juno's test code wasn't kept up to date with strangelove's interchaintest code, which was changed to follow the cosmos-sdk conventions
3) juno's test code is now updated 
4) that might not really matter for the stability of the tests
5) you should have a "run failed test again" option as a repo admin.  The way tests passed previously is that it was clicked many times until all of the tests passed

....so that's what is up with the failed test

**Comment by faddat** [comment]
* ictest-basic
* ictest-globalfee
* ictest-upgrade

**Comment by faddat** [comment]
@JakeHartnell - wrt this pr and that test:

* welp, interchaintest did not work to begin with
* then I updated it
* then the tests needed to be updated
* now idk if I broke stuff or not

Maybe just figure that:

* the merge of this PR signals the release of juno v23
* v24 is sdk 50

**Comment by faddat** [comment]
@JakeHartnell I have no idea if these work or ever worked better than clicking the button for em 20 times.  I will try and check some more.

**Comment by faddat** [comment]
* basic
* pfm
* cwhooks

**Comment by faddat** [comment]
cwhooks passes locally but not in ci

**Comment by faddat** [comment]
this is good to go, guess need to either add a delay to the cwhooks test (something with firing up the image) or just run it locally.

File: interchaintest/chain_upgrade_test.go [hunk 1]
```diff
@@ -157,7 +163,7 @@ func ValidatorVoting(t *testing.T, ctx context.Context, chain *cosmos.CosmosChai
 	require.Equal(t, haltHeight, height, "height is not equal to halt height")
 }
 
-func SubmitUpgradeProposal(t *testing.T, ctx context.Context, chain *cosmos.CosmosChain, user ibc.Wallet, upgradeName string, haltHeight uint64) string {
```
Comment (dimiandre) [inline interchaintest/chain_upgrade_test.go:160]: why switching from uint to int? 

i never seen negative blocks

**Comment by dimiandre** [comment]
ci passed now. 

instead of re-run all tests, re running only failed ones gives an higher success rate

we should migrate away from ict

**Review by dimiandre — APPROVED** [review]

**Pull request merged**